utils/db.js

```diff
@@ -1,9 +1,32 @@
 const pg = require('pg');
+const Pool = require('pg-pool');
+const url = require('url');
 
 const config = require('../config');
+const returnArgument = require('./return-argument');
+
+const postgresParams = url.parse(config.db.postgres);
+
+const auth = postgresParams.auth && postgresParams.auth.split(':');
+const poolConfig = {
+  user: auth && auth[0],
+  password: auth && auth[1],
+  host: postgresParams.hostname,
+  port: postgresParams.port,
+  database: postgresParams.pathname.split('/')[1],
+  ssl: process.env.NODE_ENV === 'production' || process.env.NODE_ENV === 'staging'
+};
+
+const pool = new Pool(poolConfig);
+
+const createClient = () => new pg.Client(config.db.postgres);
 
-const createClient = () =>
-  new pg.Client(config.db.postgres);
+const query = (sql, params, parseRow = returnArgument) =>
+  pool.query(sql, params)
+  .then((data) => (data.rows || []).map((row) => parseRow(row)))
+  .catch((err) => {
+    throw err;
```

Comment:
This catch handler is redundant - throwing in a promise will cause a rejection, so this is catching just to pass through the rejection from before - you can take this out and it will still propagate